test/gateway/eip20_cogateway/progress_redeem_with_proof.js

```diff
@@ -0,0 +1,539 @@
+    await eip20CoGateway.setStorageRoot(blockNumber, storageRoot);
+
+    blockNumber = new BN(proofData.gateway.confirm_redeem_intent.proof_data.block_number, 16);
+    storageRoot = proofData.gateway.confirm_redeem_intent.proof_data.storageHash;
+    await eip20CoGateway.setStorageRoot(blockNumber, storageRoot);
+  };
+
+  beforeEach(async function () {
+
+    owner = accounts[0];
+    facilitator = accounts[1];
+
+    redeemParams = {
+      redeemer: proofData.co_gateway.redeem.params.redeemer,
+      amount: new BN(proofData.co_gateway.redeem.params.amount, 16),
+      beneficiary: proofData.co_gateway.redeem.params.beneficiary,
+      gasPrice: new BN(proofData.co_gateway.redeem.params.gasPrice, 16),
+      gasLimit: new BN(proofData.co_gateway.redeem.params.gasLimit, 16),
+      nonce: new BN(proofData.co_gateway.redeem.params.nonce, 16),
+      hashLock: proofData.co_gateway.redeem.params.hashLock,
+      messageHash: proofData.co_gateway.progress_redeem.params.messageHash,
+      rlpParentNodes: proofData.gateway.confirm_redeem_intent.proof_data.storageProof[0].serializedProof,
+      blockHeight: new BN(proofData.gateway.confirm_redeem_intent.proof_data.block_number, 16),
+      messageStatus: MessageStatusEnum.Declared
+    };
+
+    utilityToken = await MockUtilityToken.new(
+      accounts[9], // Token address.
+      "DMY", // Symbol.
+      "Dummy token", // Token name.
+      18, // Token decimal.
+      accounts[2], // Organisation address.
+    );
+
+    bountyAmount = new BN(proofData.co_gateway.constructor.bounty);
+
+    eip20CoGateway = await EIP20CoGateway.new(
+      proofData.co_gateway.constructor.valueToken,
+      utilityToken.address,
+      proofData.co_gateway.constructor.stateRootProvider,
+      bountyAmount,
+      proofData.co_gateway.constructor.organization,
+      proofData.co_gateway.constructor.gateway,
+      proofData.co_gateway.constructor.burner,
+    );
+
+    let redeemIntentHash = coGatewayUtils.hashRedeemIntent(
+      redeemParams.amount,
+      redeemParams.beneficiary,
+      proofData.contracts.coGateway,
+    );
+
+    await eip20CoGateway.setMessage(
+      redeemIntentHash,
+      redeemParams.nonce,
+      redeemParams.gasPrice,
+      redeemParams.gasLimit,
+      redeemParams.redeemer,
+      redeemParams.hashLock,
+    );
+
+    await eip20CoGateway.setRedeem(
+      redeemParams.messageHash,
+      redeemParams.beneficiary,
+      redeemParams.amount
+    );
+
+    await eip20CoGateway.setOutboxStatus(
+      redeemParams.messageHash,
+      MessageStatusEnum.Declared,
+    );
+
+    // Set co-gateway to owner so that increase supply can be called.
+    await utilityToken.setCoGatewayAddress(owner);
+    // Send redeem amount to co-gateway.
+    await  utilityToken.increaseSupply(eip20CoGateway.address, redeemParams.amount, {from: owner});
+    // Send bounty to co-gateway.
+    await web3.eth.sendTransaction(
+      {to: eip20CoGateway.address, from: facilitator, value: bountyAmount}
```

Comment:
Also more readable when split.
```suggestion
      {
        to: eip20CoGateway.address,
        from: facilitator,
        value: bountyAmount,
      },
```